openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -509,10 +164,16 @@ def is_enabled(self, course_key=None):  # pylint: disable=arguments-differ
                 outside the context of any course.
         """
         if course_key:
-            assert isinstance(course_key, CourseKey), (
-                "Provided course_key '{}' is not instance of CourseKey.".format(course_key)
+            assert isinstance(
+                course_key, CourseKey
+            ), "Provided course_key '{}' is not instance of CourseKey.".format(
+                course_key
             )
-        return self.waffle_namespace.is_flag_active(
-            self.flag_name,
-            check_before_waffle_callback=self._get_course_override_callback(course_key),
```

Comment:
I'm going to need to do some thinking about this.  It is really important that this PR is backward compatible, or we might cause some major issues in Production. There was a reason for `check_before_waffle_callback`, and I think it had to do with getting the caching to work appropriately.  Maybe the fact that caching was moved out of the edx-toggles code resulted in this?

Reply:
In my understanding, this `check_before_waffle_callback` was only used by `CourseWaffleFlag` and `DefaultTrueWaffleFlagNamespace` (I [searched Github](https://github.com/search?q=org%3Aedx+check_before_waffle_callback&type=code)) . I made the change backward-compatible by using inheritance, in a way which is (IMHO) much cleaner.

Reply:
I still need time to review, but if the old callback was not a required solution, then it was not a great solution. :)  I will review more deeply to check that all seems well.  Thanks.